### PR TITLE
Always return counters and counter_values

### PR DESCRIPTION
**Pull request by agrare**
Always return two hashes from perf_capture_metrics because the calling
method expects it.

Catching an exception and logging the error without returning the proper
values leads to:
```
Problem collecting metrics for cu-24x7/Automation. Region [eastus2] may not be supported.
[NoMethodError]: undefined method `[]' for true:TrueClass
```

Now we'll just get the following

```
Problem collecting metrics for cu-24x7/Automation. Region [eastus2] may not be supported.
Capture for ManageIQ::Providers::Azure::CloudManager::Vm name: [cu-24x7], id: [17], start_time: [2019-08-23 00:00:00 UTC]...Complete
Skipping processing for ManageIQ::Providers::Azure::CloudManager::Vm name: [cu-24x7], id: [17], start_time: [2019-08-23 00:00:00 UTC] as no metrics were captured.
```

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1744845

### CODE REVIEW

**Comment by Ladas** [comment]
@agrare you'll need to fix also the specs, it's testing nil is returned :-)

**Comment by agrare** [comment]
Ah of course it is haha thanks @Ladas

File: spec/models/manageiq/providers/azure/cloud_manager/metrics_capture_spec.rb [hunk 1]
```diff
@@ -24,28 +24,36 @@
...
     end
 
     it "returns nothing if the region is not supported" do
       allow(ems).to receive(:insights?).and_return(true)
       allow(armrest_service).to receive(:rest_get).and_raise(::Azure::Armrest::BadRequestException.new('x', 'y', 'z'))
       expect($log).to receive(:warn).with(/problem collecting metrics/i)
-      expect(metric.perf_collect_metrics('whatever')).to be_nil
+      counters, values = metric.perf_collect_metrics('whatever')
+      expect(counters.blank?).to be_truthy
+      expect(values.blank?).to be_truthy
```
Comment (Ladas) [inline spec/models/manageiq/providers/azure/cloud_manager/metrics_capture_spec.rb]: might be the best to test we expect the `{}, {}` returned? Or the code works with `nil, nil` too?

Comment (agrare) [reply spec/models/manageiq/providers/azure/cloud_manager/metrics_capture_spec.rb]: :+1: done
